File: Medium/Python/852. Peak Index in a Mountain Array.py
class Solution(object):
    def peakIndexInMountainArray(self, arr):
        left = 0
        right = len(arr) - 1

        while left < right:
            temp = (left + right) / 2
            if arr[temp] < arr[temp + 1]:
                left = temp + 1
            else:
                right = temp

        return left

File: Medium/Java/852. Peak Index in a Mountain Array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int left = 0;
        int right = arr.length - 1;

        while(left < right){
            int temp = (left + right) / 2;
            if(arr[temp] < arr[temp + 1])
                left = temp + 1;
            else
                right = temp;
        }

        return left;
    }
}
